src/python/todo.py

```
@@ -162,19 +162,17 @@ def delete_task(self, task_index):
             IndexError: If the `task_index` is out of range for the `self.tasks` list.
             RuntimeError: If a database error or timeout occurs.
         """
-        if 0 <= task_index < len(self.tasks):
+        with log_context(log, "delete_task", operation_id=self.generate_operation_id()):
+            if not (0 <= task_index < len(self.tasks)):
+                log.error("Invalid task index [TaskIndex: %d]", task_index)
+                raise IndexError("Invalid task index!")
+
             task_id = self.tasks[task_index][0]
             try:
                 task = self.db.get_task(task_id)
```

Ruff (0.8.2)

Local variable `task` is assigned to but never used

Remove assignment to unused variable `task`

(F841)